### PR TITLE
qt/6: Bump openssl version

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **qt/6.x.x**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/qt/6.x.x/conanfile.py [hunk 1]
```diff
@@ -279,7 +279,7 @@ def validate(self):
     def requirements(self):
         self.requires("zlib/1.2.11")
```
Comment (ericLemanissier) [inline recipes/qt/6.x.x/conanfile.py:280]:
```suggestion
        self.requires("zlib/1.2.12")
```

File: recipes/qt/6.x.x/conanfile.py [hunk 2]
```diff
@@ -279,7 +279,7 @@ def validate(self):
...
         if self.options.openssl:
-            self.requires("openssl/1.1.1m")
+            self.requires("openssl/1.1.1n")
         if self.options.with_pcre2:
             self.requires("pcre2/10.37")
```
Comment (ericLemanissier) [inline recipes/qt/6.x.x/conanfile.py:284]:
```suggestion
            self.requires("pcre2/10.39")
```

**Comment by jgsogo** [comment]
I've checked it and `harfbuzz/4.2.0` is available for those settings: https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/prod/TapaholesList/14/summary.json, but it computes a different packageID.

We need to know where does this packageID for `harfbuzz` comes from (`1b94dda612e9a43ab30610d3a5a3f4483788d3a1`), it should be some recipe modifying a default option value, or overriding some major version of some requirement.

**Review by AndreyMlashkin — APPROVED** [review]

File: recipes/qt/6.x.x/conanfile.py [hunk 3]
```diff
@@ -277,11 +277,11 @@ def validate(self):
...
 
     def requirements(self):
-        self.requires("zlib/1.2.11")
+        self.requires("zlib/1.2.12")
         if self.options.openssl:
-            self.requires("openssl/1.1.1m")
+            self.requires("openssl/1.1.1n")
         if self.options.with_pcre2:
-            self.requires("pcre2/10.37")
+            self.requires("pcre2/10.39")
```
Comment (ericLemanissier) [inline recipes/qt/6.x.x/conanfile.py:284]:
```suggestion
            self.requires("pcre2/10.37")
```
[this](https://github.com/conan-io/conan-center-index/commit/912758d0b9a95965c62b79d96856511126700ccf#diff-a07f33d6e09514bd2cec71f0c02f6991f85961b406df0b566f8c687f19995804R194-R195) breaks the detection of pcre2 by qt: https://github.com/qt/qtbase/blob/v6.2.4/cmake/FindWrapSystemPCRE2.cmake#L27-L32

**Review by AndreyMlashkin — APPROVED** [review]

**Comment by PatSche** [comment]
Since approx. 2 days, we have 21 pending checks. Is this usual, or are there any problems with your CI system?

**Comment by SpaceIm** [comment]
you can also bump icu, libpq & expat.

**Comment by ericLemanissier** [comment]
@MartinDelille please consider https://github.com/MartinDelille/conan-center-index/pull/7

**Review by SpaceIm — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by AndreyMlashkin — APPROVED** [review]

**Review by AndreyMlashkin — APPROVED** [review]

File: recipes/qt/6.x.x/conanfile.py [hunk 4]
```diff
@@ -277,27 +277,27 @@ def validate(self):
...
```
Comment (ericLemanissier) [inline recipes/qt/6.x.x/conanfile.py:284]:
```suggestion
            self.requires("pcre2/10.37") # needs to be < 10.38
```
Woopsie, we already went through this one earlier in this PR 😊

Comment (SpaceIm) [reply recipes/qt/6.x.x/conanfile.py:284]: Due to https://github.com/PhilipHazel/pcre2/commit/2410fbe3869cab403f02b94caa9ab37ee9f5854b

I've left a comment few weeks ago to warn against this breaking change in static lib name for Visual Studio: https://github.com/PhilipHazel/pcre2/commit/2410fbe3869cab403f02b94caa9ab37ee9f5854b#commitcomment-68176307

**Comment by ericLemanissier** [comment]
@MartinDelille I integrated all your changes in https://github.com/conan-io/conan-center-index/pull/10261 so you can close this PR if you want.

**Comment by MartinDelille** [comment]
Thanks @ericLemanissier ! I close it in favor of the promising https://github.com/conan-io/conan-center-index/pull/10261 then! 😄

**Pull request closed without merging**